File: spec/factories/levels.rb
# == Schema Information
#
# Table name: levels
#
#  id         :integer          not null, primary key
#  exp_min    :integer
#  exp_max    :integer
#  level      :integer
#  created_at :datetime         not null
#  updated_at :datetime         not null
#
# Indexes
#
#  index_levels_on_level  (level)
#

FactoryGirl.define do
  factory :level do
    exp_min 1
    exp_max 1
    level 1
  end
end

File: packages/db/src/schema/levels.ts
import {
  pgTable,
  serial,
  integer,
  timestamp,
  index,
} from "drizzle-orm/pg-core";

export const levels = pgTable(
  "levels",
  {
    id: serial("id").primaryKey(),
    level: integer("level").notNull(),
    expMin: integer("exp_min").notNull(),
    expMax: integer("exp_max").notNull(),
    createdAt: timestamp("created_at").notNull().defaultNow(),
    updatedAt: timestamp("updated_at").notNull().defaultNow(),
  },
  (table) => [index("levels_level_idx").on(table.level)]
);
